### PR TITLE
Fix docs for memory-swap

### PR DESCRIPTION
**Pull request by hqhq**
Fixes: #18894

Signed-off-by: Qiang Huang h.huangqiang@huawei.com

### CODE REVIEW

**Comment by thaJeztah** [comment]
LGTM, thanks @hqha

**Comment by thaJeztah** [comment]
ping @moxiegirl for review and merge

File: api/client/build.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	pull := cmd.Bool([]string{"-pull"}, false, "Always attempt to pull a newer version of the image")
 	dockerfileName := cmd.String([]string{"f", "-file"}, "", "Name of the Dockerfile (Default is 'PATH/Dockerfile')")
 	flMemoryString := cmd.String([]string{"m", "-memory"}, "", "Memory limit")
-	flMemorySwap := cmd.String([]string{"-memory-swap"}, "", "Total memory (memory + swap), '-1' to disable swap")
+	flMemorySwap := cmd.String([]string{"-memory-swap"}, "", "Total memory (memory + swap), '-1' to disable swap limit")
```
Comment (moxiegirl) [inline api/client/build.go]: `-1` to enable unlimited swap

File: docs/reference/commandline/build.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ parent = "smn_cli"
       --help                          Print usage
       --isolation=""                  Container isolation technology
       -m, --memory=""                 Memory limit for all build containers
-      --memory-swap=""                Total memory (memory + swap), `-1` to disable swap
+      --memory-swap=""                Total memory (memory + swap), `-1` to disable swap limit (unlimited)
```
Comment (moxiegirl) [inline docs/reference/commandline/build.md]:
```markdown
An positive integer equal to memory + swap. Specify -1 to enable unlimited swap.
```

File: docs/reference/commandline/create.md [hunk 3]
```diff
@@ -57,7 +57,7 @@ Creates a new container.
       -m, --memory=""               Memory limit
       --mac-address=""              Container MAC address (e.g. 92:d0:c6:0a:29:33)
       --memory-reservation=""       Memory soft limit
-      --memory-swap=""              Total memory (memory + swap), '-1' to disable swap
+      --memory-swap=""              Total memory (memory + swap), '-1' to disable swap limit (unlimited)
```
Comment (moxiegirl) [inline docs/reference/commandline/create.md]:
```markdown
An positive integer equal to memory plus swap. Specify -1 to enable 
unlimited swap.
```

File: docs/reference/commandline/run.md [hunk 4]
```diff
@@ -56,7 +56,7 @@ parent = "smn_cli"
       -m, --memory=""               Memory limit
       --mac-address=""              Container MAC address (e.g. 92:d0:c6:0a:29:33)
       --memory-reservation=""       Memory soft limit
-      --memory-swap=""              Total memory (memory + swap), '-1' to disable swap
+      --memory-swap=""              Total memory (memory + swap), '-1' to disable swap limit (unlimited)
```
Comment (moxiegirl) [inline docs/reference/commandline/run.md]:
```markdown
An positive integer equal to memory plus swap. Specify -1 to enable 
unlimited swap.
```

File: man/docker-build.1.md [hunk 5]
```diff
@@ -93,7 +93,7 @@ set as the **URL**, the repository is cloned locally and then sent as the contex
   Memory limit
 
 **--memory-swap**=*MEMORY-SWAP*
```
Comment (moxiegirl) [inline man/docker-build.1.md]:
```markdown
MEMORY-SWAP is positive integer equal to memory plus swap. Specify -1 to enable 
unlimited swap.
```

**Comment by hqhq** [comment]
@moxiegirl Updated, thanks.

**Comment by hqhq** [comment]
@thaJeztah @moxiegirl I mistakenly deleted this branch and after restore, I can reopen this PR, can you help reopen it or do I have to open a new one?

File: man/docker-run.1.md [hunk 6]
```diff
@@ -330,7 +330,7 @@ as memory limit.
 **--memory-swap**=""
```
Comment (moxiegirl) [inline man/docker-run.1.md]:
```markdown
**--memory-swap**="SWAP_LIMIT"
   A limit value equal to memory plus swap. Must be used with the  **-m**
   (**--memory**) flag. This value should always larger than **-m**
   (**--memory**) value. 

   The format of `SWAP_LIMIT` is <`number`>[<`unit`>]. Unit can be `b` (bytes),
   `k` (kilobytes), `m` (megabytes), or `g` (gigabytes). If you don't specify a
   unit, `m` is used. Set SWAP_LIMIT to `-1` to enable unlimited swap.
```

File: man/docker-create.1.md [hunk 7]
```diff
@@ -233,7 +233,7 @@ as memory limit.
 **--memory-swap**=""
```
Comment (moxiegirl) [inline man/docker-create.1.md]:
```markdown
**--memory-swap**="SWAP_LIMIT"
   A limit value equal to memory plus swap. Must be used with the  **-m**
   (**--memory**) flag. This value should always larger than **-m**
   (**--memory**) value. 

   The format of `SWAP_LIMIT` is <`number`>[<`unit`>]. Unit can be `b` (bytes),
   `k` (kilobytes), `m` (megabytes), or `g` (gigabytes). If you don't specify a
   unit, `m` is used. Set SWAP_LIMIT to `-1` to enable unlimited swap.
```

**Comment by moxiegirl** [comment]
@hqhq sorry to come in with late changes...I was in the middle of my review.  Two things these last changes assume:

* That user _must_ specify `-m` with this flag --- is that true?
* That if a user does not specify a unit we default to megabytes --- is that true?

**Comment by hqhq** [comment]
> That user must specify -m with this flag --- is that true?

Yes it's true, because `--memory-swap` must be larger than `-m`, while not assigning `-m` means unlimited memory, which can not be larger.

> That if a user does not specify a unit we default to megabytes --- is that true?

No, if a user does not specify a unit, we default to bytes.

**Comment by hqhq** [comment]
@moxiegirl Is this PR can be reopened? I can't push any changes unless it's reopened. Can you help and confirm? Thanks.

**Comment by hqhq** [comment]
> Notice that I'm proposing **--memory-swap**="SWAP_LIMIT" in the flag; but I think that work LIMIT is important. You could also do `--memory-swap="LIMIT" which is shorter but also makes clear what the value is doing. WDYT?

Yeah I think `LIMIT` or `MEMORY_SWAP` is better, I'll take `LIMIT`, thanks.

**Comment by moxiegirl** [comment]
@hqhq I don't have the perms to reopen for some reason. @thaJeztah should be able to.  I'll also created a carry branch --- just in case: https://github.com/moxiegirl/docker/tree/carry-18902

File: runconfig/parse.go [hunk 8]
```diff
@@ -93,7 +93,7 @@ func Parse(cmd *flag.FlagSet, args []string) (*container.Config, *container.Host
 		flHostname          = cmd.String([]string{"h", "-hostname"}, "", "Container host name")
 		flMemoryString      = cmd.String([]string{"m", "-memory"}, "", "Memory limit")
 		flMemoryReservation = cmd.String([]string{"-memory-reservation"}, "", "Memory soft limit")
-		flMemorySwap        = cmd.String([]string{"-memory-swap"}, "", "Total memory (memory + swap), '-1' to disable swap")
+		flMemorySwap        = cmd.String([]string{"-memory-swap"}, "", "Total memory (memory + swap), '-1' to disable swap limit")
```
Comment (moxiegirl) [inline runconfig/parse.go]: see comment on build.go file

**Comment by thaJeztah** [comment]
Sorry, looks like I'm not able to reopen as well :cry:

**Comment by hqhq** [comment]
Handled by #18964
Thanks for you confirm, must be a github bug on this.

**Pull request closed without merging**